Fix clone object with extrafield type "object link"

aconits opened the pull request:
# Fix

With extrafield type "object to link" we can't clone correctly 
Basically when we editing card the value sent is equal to "ref" but when we clone we try to insert by sending the id instead of "ref"

eldy commented:
I suggest a more complete fix:
When is_numeric($value) is false, there is no need to make the first fetch, can you enhance your change to avoid duplicate fetch when we can ?

eldy committed:
Fix align

eldy committed:
Merge branch '4.0' of git@github.com:Dolibarr/dolibarr.git into 4.0

Commit:
Merge remote-tracking branch 'upstream/4.0' into 4.0

Commit:
Merge remote-tracking branch 'upstream/3.9' into 3.9

Commit:
Fix #5582 Can't add or change customer ref on proposal

eldy committed:
Merge pull request #5576 from atm-florian/3.9_fixPHPStrict

FIX php Strict

eldy committed:
Merge pull request #5578 from aspangaro/4.0-p37

Uniformize presentation

eldy committed:
Merge pull request #5580 from simnandez/4.0

NEW Add Panama datas

eldy committed:
Merge pull request #5583 from atm-florian/3.9

FIX : update limit stock on product stock

eldy committed:
Fix colspan

eldy committed:
Merge branch '4.0' of git@github.com:Dolibarr/dolibarr.git into 4.0

eldy committed:
Merge branch '3.9' of git@github.com:Dolibarr/dolibarr.git into 4.0

Conflicts:
	htdocs/core/modules/barcode/modules_barcode.class.php
	htdocs/product/admin/product.php

eldy committed:
Fix bad link

eldy committed:
Fix thumbs are using same extension than original.

eldy committed:
Fix code comment

Commit:
FIX #5594

Commit:
better test

Commit:
revert better fix

eldy committed:
Work on module website

Conflicts:
	htdocs/websites/index.php

Commit:
Work on accoutancy PgSQL comptatiblity and other stuff

Commit:
fix list

eldy committed:
Fix edit of page in module website

Conflicts:
	htdocs/websites/class/website.class.php

eldy committed:
Fix code generator

aspangaro committed:
Uniformize presentation

eldy committed:
Fix several errors in search an sorting in stock by lot view

eldy committed:
Fix list of lot

eldy committed:
Merge pull request #5595 from atm-florian/FIX_5594

FIX #5594

eldy committed:
Merge pull request #5596 from atm-florian/40_accountancy

40 accountancy

Commit:
better fix for #5594

Commit:
Merge branch '3.9' of https://github.com/Dolibarr/dolibarr.git into FIX_5594

eldy committed:
Fix missing supplier proposals in project overview

eldy committed:
Merge branch '4.0' of git@github.com:Dolibarr/dolibarr.git into 4.0

aspangaro committed:
Uniformize presentation

aspangaro committed:
Merge remote-tracking branch 'Upstream/4.0' into 4.0-p37

aspangaro committed:
FIX : Accountancy - BANK_DISABLE_DIRECT_INPUT Add an option

Commit:
FIX : MULTICURRENCY if value is 1,56 then GETPOST 'int' check return ''

Commit:
FIX : create proposal from project

Commit:
fix for finche inter same problem

aspangaro committed:
Uniformize presentation on opensurvey

aspangaro committed:
Fix: Missing language file for user export when the module members is on

aspangaro committed:
Fix: Missing language key for bank export

aspangaro committed:
Fix: MIssing language file for category members export

aspangaro committed:
Fix: Missing Language file and key for category contact export

aspangaro committed:
Uniformize presentation on contact

aspangaro committed:
Fix: Typo on invoice export

aspangaro committed:
Fix: Missing language file for salaries export

aspangaro committed:
Uniformize presentation on categories

Commit:
fix export

Commit:
force export txt for ciel and quadratus

aspangaro committed:
Fix: Accountancy - Problem to define account_parent for an accounting account & presentation

aspangaro committed:
Replace with 0

eldy committed:
Fix look and feel

eldy committed:
Fix translation

eldy committed:
Fix several errors with bank account module enabled on sale tax edition

eldy committed:
Another fix on bank account missing when bank module enabled.

Commit:
Merge pull request #5600 from atm-florian/FIX_5594

Fix 5594

Commit:
Merge pull request #5602 from aspangaro/4.0-p38

FIX : Accountancy - BANK_DISABLE_DIRECT_INPUT Add an option

Commit:
Merge pull request #5603 from atm-florian/4.0_FIX_Multicurrency_with_french_number_setting

FIX : MULTICURRENCY if value is 1,56 then GETPOST 'int' check return ''

Commit:
Merge pull request #5604 from atm-florian/FIX_createpropalfromproject

FIX : create proposal from project

Commit:
Merge pull request #5615 from aspangaro/4.0-p41

Fix: Accountancy - Problem to define account_parent for an accounting account + presentation

Commit:
Merge pull request #5613 from atm-florian/40_accountancy

fix export

fappels committed:
Fix sellby without hour check

fappels committed:
Fix SQL error insert supplier order line

$this->fk_multicurrency not set gives SQL error

aspangaro committed:
Uniformize presentation, add button cancel everywhere

Commit:
FIX: #5594

Commit:
FIX : #5629 PgSQL Interger string stylish error

Commit:
FIX : #5629 DoliDBPgsql::query SQL Error message: ERROR: 22P02: invalid
input syntax for integer: ""

joseplluis committed:
Bug: delivery date value is not hide with global ORDER_DISABLE_DELIVERY_DATE

In title and in filter the delivery date column is well hidden. 
The error is only in column value,

Commit:
FIX : in PgSQL no quote "word style" is permitted around column name

eldy committed:
Fix closing td

eldy committed:
Merge branch '4.0' of git@github.com:Dolibarr/dolibarr.git into 4.0

eldy committed:
Fix order of module for demo.

eldy committed:
Fix responsive design on small smartphone

eldy committed:
Fix fiter on contract date

eldy committed:
Merge pull request #5635 from joseplluis/patch-3

Fix Bug: delivery date value is not hide with global ORDER_DISABLE_DELIVE…

eldy committed:
Merge pull request #5631 from atm-florian/4.0

FIX #5594

Commit:
Merge branch '3.9' of https://github.com/Dolibarr/dolibarr.git into 3.9

Commit:
FIX : PgSQL Module Ressource list crash #5637

Commit:
another fix for Ressource module in PgSQL

Commit:
Merge branch '4.0' of https://github.com/Dolibarr/dolibarr.git into 4.0

Commit:
other fix for PgSQL and style

Commit:
fix ressource for PGSQL

eldy committed:
Merge pull request #5601 from aspangaro/4.0-p37

Uniformize title presentation on product/service card

eldy committed:
Merge pull request #5605 from aspangaro/4.0-p39

Uniformize presentation on opensurvey

eldy committed:
Merge pull request #5608 from aspangaro/4.0-p40

Fix: Some problems on export with language

eldy committed:
Merge pull request #5618 from fappels/4.0_fix_eatby_sellby_check

Fix sellby without hour check

eldy committed:
Merge pull request #5619 from fappels/4.0_fix_sql_error_insert_supplier_order_line

Fix SQL error insert supplier order line

eldy committed:
Merge pull request #5625 from aspangaro/4.0-p42

Uniformize presentation, add button cancel everywhere

eldy committed:
Merge pull request #5630 from atm-florian/3.9

FIX : #5629 PgSQL Interger string stylish error

eldy committed:
Merge pull request #5638 from atm-florian/3.9_fix_5637

FIX : PgSQL Module Ressource list crash #5637

eldy committed:
Merge remote-tracking branch 'origin/3.9' into 4.0

Conflicts:
	htdocs/commande/list.php
	htdocs/core/class/commonobject.class.php
	htdocs/core/class/extrafields.class.php
	htdocs/resource/class/resource.class.php

Commit:
Merge branch '4.0' of https://github.com/Dolibarr/dolibarr.git into 4.0

Conflicts:
	htdocs/resource/class/dolresource.class.php

eldy committed:
Fix no way to fix the ref of recurring invoice when an error was done
into typing.

grandoc committed:
fix : translation

grandoc committed:
fix : bad sort

eldy committed:
Fix cancel button ignored

eldy committed:
Fix date of contract not used

eldy committed:
Code comment

eldy committed:
CSS

Commit:
Merge branch '4.0' of https://github.com/Dolibarr/dolibarr.git into 4.0

hregis committed:
Fix: wrong "lastoutput" and "lastresult" if jobtype is "method" or
"function"

eldy committed:
Fix wrapping with jmobile

Commit:
Merge branch '4.0' of https://github.com/Dolibarr/dolibarr.git into 4.0

Commit:
Merge pull request #5644 from grandoc/3.9

fix : translation

Commit:
Merge pull request #5645 from grandoc/4.0

fix : bad sort

Commit:
Merge pull request #5639 from atm-florian/4.0

fix ressource module  for PGSQL and coding style

Commit:
Merge pull request #5648 from hregis/3.8_bug

Fix: wrong "lastoutput" and "lastresult" if jobtype is "method" or "function"

aspangaro committed:
Fix: Presentation of Donations Area and add search

eldy committed:
Merge pull request #5662 from aspangaro/4.0-p43

Fix: Presentation of Donations Area and add search

eldy committed:
Solve php regression failing travis

eldy committed:
Merge branch '3.8' of git@github.com:Dolibarr/dolibarr.git into 3.9

eldy committed:
Merge branch '3.9' of git@github.com:Dolibarr/dolibarr.git into 4.0

Conflicts:
	htdocs/cron/class/cronjob.class.php
	htdocs/install/pgsql/functions/functions.sql

eldy committed:
Merge branch '4.0' of git@github.com:Dolibarr/dolibarr.git into 4.0

eldy committed:
Fix field customer/prospect/supplier in list of thirdparties

eldy committed:
Fix for jmobile

eldy committed:
Fix for jmobile

eldy committed:
Fix responsive design

eldy committed:
CSS

eldy committed:
Fix not used field

eldy committed:
FIX usage of sepa generation setup.

eldy committed:
Release 4.0

eldy committed:
Prepare 4.0

eldy committed:
Fix sort of list and number of records

eldy committed:
Fix sort of fields

aspangaro committed:
Fix: Wrong language key

eldy committed:
Fix for jmobile

eldy committed:
Fix missing /form

eldy committed:
Fix css

eldy committed:
Fix css

eldy committed:
Merge pull request #5666 from aspangaro/4.0-p45

Fix: Wrong language key

eldy committed:
Fix css

eldy committed:
Merge branch '4.0' of git@github.com:Dolibarr/dolibarr.git into 4.0

eldy committed:
Fix lang not loaded

eldy committed:
Merge branch '4.0' of git@github.com:Dolibarr/dolibarr.git into 4.0

aspangaro committed:
Fix: VATArea - Problem of presentation

eldy committed:
Fix sort on amount

eldy committed:
Fix several fix into tab management of tasks

eldy committed:
Fix missing filter

Commit:
Merge remote-tracking branch 'upstream/3.9' into 3.9

Commit:
Merge pull request #5675 from aspangaro/4.0-p46

Fix: VAT Area - Problem of presentation

Commit:
Fix #5641 Dolibarr 4.0 RC2 bug when create new contract

Commit:
Merge remote-tracking branch 'upstream/4.0' into 4.0

Commit:
Merge pull request #5681 from simnandez/4.0

Fix #5641 Dolibarr 4.0 RC2 bug when create new contract

atm-gauthier committed:
FIX : header title in commercial area

eldy committed:
FIX Several problems when using smartphone

eldy committed:
Merge branch '4.0' of git@github.com:Dolibarr/dolibarr.git into 4.0

eldy committed:
Fix responsive design

eldy committed:
Fix another field not responsive

eldy committed:
Fix Quick hack to solve legal problem in some countries.

eldy committed:
Fix sql error

eldy committed:
Fix hrm module

eldy committed:
Fix var $conf not defined

eldy committed:
Fix var $conf not defined

eldy committed:
Fix button create contract from order not visible sometimes

eldy committed:
Fix sort on wrong field

eldy committed:
FIX Missing filter on environment

grandoc committed:
internationalized comments

aspangaro committed:
Fix: Info tab on product card - Double header

fappels committed:
Ignore time when checking lot dates

fmarcet committed:
Fix: An empty variable is used to update multiprice

Commit:
Merge pull request #5688 from grandoc/4.0

Fix: internationalized comments

Commit:
Merge pull request #5690 from aspangaro/4.0-p47

Fix: Info tab on product card - Double header

Commit:
Fix lost value if use double quote

fappels committed:
Delete product_lot when product deleted

eldy committed:
Merge pull request #5691 from fappels/3.9_fix_check_lot_dates

FIX #5642 Ignore time when checking lot dates

eldy committed:
Merge pull request #5692 from fmarcet/3.9

Fix: An empty variable is used to update multiprice

eldy committed:
Merge pull request #5685 from atm-gauthier/3.8_fix_header_title_commercial_area

FIX : header title in commercial area

eldy committed:
Merge pull request #5694 from atm-ph/fix_3.8_address_empty

Fix lost value if use double quote

eldy committed:
Merge pull request #5695 from fappels/4.0_delete_product_lot

FIX Delete product_lot data when product deleted

eldy committed:
Merge branch '3.8' of git@github.com:Dolibarr/dolibarr.git into 3.9

eldy committed:
Merge branch '3.9' of git@github.com:Dolibarr/dolibarr.git into 3.9

eldy committed:
Better fix for 416a0892d75438a4083c94176d803a5c723d8fa9

eldy committed:
Merge branch '3.9' of git@github.com:Dolibarr/dolibarr.git into 4.0

Conflicts:
	htdocs/product/price.php
	htdocs/product/stock/class/mouvementstock.class.php

eldy committed:
Merge branch '4.0' of git@github.com:Dolibarr/dolibarr.git into 4.0

eldy committed:
FIX #5687

eldy committed:
FIX #5674

eldy committed:
FIX #5651

eldy committed:
FIX #5660

eldy committed:
sql syntax

eldy committed:
FIX #5589

eldy committed:
Merge branch '3.9' of git@github.com:Dolibarr/dolibarr.git into 4.0

Conflicts:
	htdocs/product/price.php

eldy committed:
FIX quick search boxes when nojs or when using smartphone

eldy committed:
Fix bad translation

eldy committed:
Fix css

eldy committed:
Fix http into https

Commit:
Merge remote-tracking branch 'upstream/3.9' into 3.9

Commit:
Merge remote-tracking branch 'origin/3.9' into 3.9

Commit:
FIX: vat dictionary should allow enter and edit multiple values for localtaxes, separated by: (ex -19:-15)

hregis committed:
Fix: Warning: preg_match(): No ending delimiter '/' found

fappels committed:
Merge remote-tracking branch 'refs/remotes/Dolibarr/4.0' into 4.0_debug_product_lot

fappels committed:
Replace non existing ref parameter

Replace non existing ref parameter with product_id and batch to fetch a
specific batch from a product.
Remove unused sceleton methods.

fappels committed:
Fix insert expeditiondet_batch with extrafield

Line id of extraflied was used iso expeditiondet

Commit:
Merge pull request #5703 from hregis/4.0_bug

Fix: Warning: preg_match(): No ending delimiter '/' found

sanser committed:
fileconf.php redundant else statements

Is it ok the 2 else blocks are the same?

aspangaro committed:
Fix: Problem of presentation with index supplier command

hregis committed:
Fix: increase max-width when use externals modules

eldy committed:
FIX SEC for HTB23302

eldy committed:
Merge branch '4.0' of git@github.com:Dolibarr/dolibarr.git into 4.0

eldy committed:
Prepare 4.0.1

philippe-opendsi committed:
No initialisation of the array before filling it. May cause some trouble when fetch is call several times and lines added or deleted

eldy committed:
FIX Update of maxnbrun on job list failed.

eldy committed:
FIX Execute a dedicated job from its id may results of launching other
jobs too.

Commit:
Fix miss entity column in llx_multicurrency_rate

Commit:
Default entity 1 in llx_multicurrency_rate

grandoc committed:
fix : bad name

eldy committed:
FIX Translation of "Name" is not a good choice for floow-up.

eldy committed:
Fix late satus on contract service line

Commit:
Merge pull request #5702 from simnandez/3.8

FIX: vat dictionary should allow enter and edit multiple values for localtaxes, separated by: (ex -19:-15)

Commit:
Merge pull request #5719 from philippe-opendsi/3.8_patch_1

FIX #5705 Supplier Order's lines array initalisation

Commit:
Merge pull request #5713 from sanser/4.0

FIX: fileconf.php redundant else blocks

eldy committed:
FIX Mandatory field payment term was not css highlighted.

Commit:
Merge pull request #5714 from aspangaro/4.0-p48

FIX: Problem of presentation with index supplier command

Commit:
Merge pull request #5722 from atm-ph/fix_4.0_multicurrency_rate_entity

Fix miss entity column in llx_multicurrency_rate

eldy committed:
Merge branch '4.0' of git@github.com:Dolibarr/dolibarr.git into 4.0

Commit:
Merge pull request #5724 from grandoc/4.0

fix : bad name

eldy committed:
Merge branch '4.0' of git@github.com:Dolibarr/dolibarr.git into 4.0

Commit:
Merge remote-tracking branch 'upstream/3.8' into 3.9

Conflicts:
	htdocs/admin/dict.php
	htdocs/fourn/class/fournisseur.commande.class.php

eldy committed:
Fix minor css

fmarcet committed:
Fix: Search all  is lost on lists when pagination or when the columns are sorted

fmarcet committed:
Fix: Search all  is lost on lists when pagination or when the columns are sorted

Commit:
Merge remote-tracking branch 'upstream/3.9' into 4.0

Conflicts:
	htdocs/fourn/class/fournisseur.commande.class.php

eldy committed:
Fix option MAIN_BUTTON_HIDE_UNAUTHORIZED

eldy committed:
Merge branch '4.0' of git@github.com:Dolibarr/dolibarr.git into 4.0

eldy committed:
Fix multicompany: recurring invoice must be in same env than template.

eldy committed:
Fix when creating recurring invoice, date next gen was not set

eldy committed:
Fix compatibility of recurring invoices with postgresql

eldy committed:
Fix

eldy committed:
More log to help debug

eldy committed:
Fix error message not reported

eldy committed:
Merge pull request #5727 from fmarcet/3.9

Fix: Search all  is lost on lists when pagination or when the columns…

eldy committed:
Merge pull request #5716 from hregis/3.9_bug

Fix: increase max-width when use externals modules

eldy committed:
Merge pull request #5710 from fappels/4.0_debug_product_lot

FIX create shipment with lot and extrafield

eldy committed:
Fix missing information on main supplier proposal card, not on note tab.

eldy committed:
FIX alignement of intervention status

eldy committed:
Fix date must be with standard format

eldy committed:
Fix missing a filter billed=0 into link of billable orders

eldy committed:
FIX inversion customer/supplier price

eldy committed:
Fix list of supplier price must show price compared to quantity.

eldy committed:
Fix #5712

eldy committed:
FIX #5734

eldy committed:
FIX Clean of search fields

eldy committed:
Fix better fix to increase size of login_block to allow external icon.

eldy committed:
Merge branch '3.9' of git@github.com:Dolibarr/dolibarr.git into 4.0

Conflicts:
	htdocs/comm/propal/list.php
	htdocs/commande/list.php
	htdocs/compta/facture/list.php
	htdocs/contact/list.php
	htdocs/contrat/list.php
	htdocs/expedition/list.php
	htdocs/expensereport/list.php
	htdocs/fichinter/list.php
	htdocs/fourn/commande/list.php
	htdocs/product/reassort.php
	htdocs/product/reassortlot.php
	htdocs/product/stock/list.php
	htdocs/societe/list.php
	htdocs/supplier_proposal/list.php
	htdocs/theme/eldy/style.css.php

eldy committed:
Fix bad merge

eldy committed:
Fix deprecated option should be on

eldy committed:
FIX #2991

eldy committed:
On non public page, access with a bad fetch id can return error

eldy committed:
Fix default value of new table

Commit:
Merge branch '4.0' of https://github.com/Dolibarr/dolibarr into fix_develop_clone_with_extra_link

Commit:
better fix to clone object with "link" as extrafield
